packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -2068,3 +2130,406 @@ class RenderNestedScrollViewViewport extends RenderViewport {
+  void unregisterState(NestedScrollViewState value) {
+    _currentState = null;
+  }
+
+  /// The [ScrollController] of the outer scroll view
+  ScrollController get outerController => _outerController;
+  /// The [ScrollController] of the inner scroll view
+  ScrollController get innerController => _innerController;
+
+  _ControllerState _internalState = _ControllerState.idle;
+
+  @override
+  bool get _floatHeaderSlivers {
+    assert(_currentState != null);
+    return _currentState!.widget.floatHeaderSlivers;
+  }
+
+  @override
+  double get offset => _outerController.offset;
+
+  @override
+  Future<void> animateTo(
+    double offset, {
+    required Duration duration,
+    required Curve curve,
+  }) async {
+    _animateTo(offset, duration: duration, curve: curve);
+  }
+
+  @override
+  void jumpTo(double value) {
+    _jumpTo(value);
+  }
+
+  @override
+  void attach(ScrollPosition position) {
+    assert(position is _NestedScrollPosition);
+    switch ((position as _NestedScrollPosition).type) {
+      case _NestedType.outer:
+        super.attach(position);
+        _outerController.attach(position);
+        _internalState = _ControllerState.outerPositionAttached;
+        break;
+      case _NestedType.inner:
+        _innerController.attach(position);
+        _internalState = _ControllerState.innerPositionAttached;
+        break;
+    }
+  }
+
+  @override
+  void detach(ScrollPosition position) {
+    assert(position is _NestedScrollPosition);
+    switch ((position as _NestedScrollPosition).type) {
+      case _NestedType.outer:
+        super.detach(position);
+        _outerController.detach(position);
+        _internalState = _outerPosition == null
+          ? _ControllerState.idle
+          : _ControllerState.outerPositionAttached;
+        break;
+      case _NestedType.inner:
+        _innerController.detach(position);
+        _internalState = _innerPositions.isEmpty
+          ? _ControllerState.outerPositionAttached
+          : _ControllerState.innerPositionAttached;
+        break;
+    }
+  }
+
+  @override
+  ScrollPosition createScrollPosition(
+    ScrollPhysics physics,
+    ScrollContext context,
+    ScrollPosition? oldPosition,
+  ) {
+    switch(_internalState) {
+      case _ControllerState.idle:
+        return _outerController.createScrollPosition(physics, context, oldPosition);
+      case _ControllerState.outerPositionAttached:
+      case _ControllerState.innerPositionAttached:
+        return _innerController.createScrollPosition(physics, context, oldPosition);
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _currentDrag?.dispose();
+    _currentDrag = null;
+    _outerController.dispose();
+    _innerController.dispose();
+  }
+
+  @override
+  String toString() => '${objectRuntimeType(this, 'NestedScrollController')}(outer=$_outerController; inner=$_innerController)';
+}
+
+class _NestedScrollBehavior implements ScrollBehavior {
```

Comment:
This seems like a separate change, unrelated to just exposing the scroll controller. What is this meant to do?